samtranslator/model/api/api_generator.py

```diff
@@ -676,10 +829,19 @@ def to_cloudformation(
         :returns: a tuple containing the RestApi, Deployment, and Stage for an empty Api.
         :rtype: tuple
         """
+        api_domain_response: Union[ApiDomainResponseV2, ApiDomainResponse]
+        domain: Union[Resource, None]
+        basepath_mapping: Union[List[ApiGatewayBasePathMapping], List[ApiGatewayBasePathMappingV2], None]
         rest_api = self._construct_rest_api()
-        api_domain_response = self._construct_api_domain(rest_api, route53_record_set_groups)
-        domain = api_domain_response.domain
-        basepath_mapping = api_domain_response.apigw_basepath_mapping_list
+        if isinstance(self.domain, dict) and self.domain.get("EndpointConfiguration") == "PRIVATE":
+            api_domain_response = self._construct_api_domain_v2(rest_api, route53_record_set_groups)
+            domain = api_domain_response.domain
+            basepath_mapping = api_domain_response.apigw_basepath_mapping_list
+        else:
+            api_domain_response = self._construct_api_domain(rest_api, route53_record_set_groups)
+            domain = api_domain_response.domain
+            basepath_mapping = api_domain_response.apigw_basepath_mapping_list
+
```

Comment:
It feels cleaner if you only separate the `api_domain_response` definition, considering the other ones are the same. And it's conceptually much easier to understand (and the diff is smaller)

```
        if isinstance(self.domain, dict) and self.domain.get("EndpointConfiguration") == "PRIVATE":
            api_domain_response = self._construct_api_domain_v2(rest_api, route53_record_set_groups)
        else:
            api_domain_response = self._construct_api_domain(rest_api, route53_record_set_groups)
        domain = api_domain_response.domain
        basepath_mapping = api_domain_response.apigw_basepath_mapping_list
```

Reply:
Updated